Tests/MetricsTests/MetricsTests.swift

```diff
@@ -95,6 +95,28 @@ class MetricsExtensionsTests: XCTestCase {
         XCTAssertEqual(UInt64(testTimer.values.first!), end.uptimeNanoseconds - start.uptimeNanoseconds, "expected value to match")
         XCTAssertEqual(metrics.timers.count, 1, "timer should have been stored")
     }
+    
+    func testTimerDuration() throws {
+        guard #available(iOS 16, macOS 13, tvOS 15, watchOS 8, *) else {
+            return
+        }
+        
+        let metrics = TestMetrics()
+        MetricsSystem.bootstrapInternal(metrics)
+        
+        let name = "timer-\(UUID().uuidString)"
+        let duration = Duration(secondsComponent: 3, attosecondsComponent: 123000000000000000)
```

Comment:
since Duration does not exist in earlier swift versions this will have to be guarded "around" this code, not just with a `guard #available` since that still makes the code compile but it won't compile on old Swifts :)

Reply:
Thank you for a review!

I wrapped the inside of the test in the `#if` check because this test is added to the array of test cases to run on Linux in an auto-generated file. It's easaier to wrap the insides of the func than to work around not adding that test case conditionally in the script.